### ISSUE TITLE
Austin Cawley-Edwards: Open Source Response

### ISSUE THREAD

**Issue by austince** [jeffThompson/CreativeProgramming1#46]
Contributing to open source projects is beneficial to both individuals and corporations. For individuals, open source can be a way to join a community and make lasting connections with people of shared interests. For companies, open source provides a level playing ground to work with each other on products they can all use. Companies like Microsoft, Facebook, and Google not only open source their own projects, but also devote millions in developer salaries to contributing to these projects. I believe that it is necessary that the foundations of software be open and free, but it can pose tough ethical problems when products that generate revenue are open to all.

For example, if Google were to open source the software it uses for its self driving cars and one of the cars were to crash and kill someone, who is to blame? There is a large list of possible answers, including the developers of the project, the Company itself, the State, or even the car, each with understandable justifications. Since Google is making money from the software, does that make it at more fault? It is difficult to point fingers when open source software goes awry. 

On a less deadly note, is it fair for a company to directly profit from the work of a community? If I were to contribute to Facebook's core infrastructure, should I make a percentage of what Facebook makes? It gets hazy as the dependencies of modern software grow. Many platforms that support core business are created, built, and maintained by teams of unpaid open source contributors. Take something like React, which is a Facebook software library for building web apps; it supports a significant portion of the modern web but has many unpaid maintainers. The line between what software directly profits companies and what is too low-level to count is hazy. I think that until profit-models change, top-level, user-facing products must be kept closed source to avoid unethical use of labor and perhaps unethical errors, but ultimately that businesses should innovate ways of making money that allow everything to be open sourced and all contributors to share equity.

On the creative-art side, open source could actually hinder creativity. Artists like Zach Lieberman have publicly denied sharing code for artwork, but are very open with their ideas and platforms they use. This could be perhaps because it would be easy for someone to take their code and pretend to be the creator. He combats this seclusion with many initiatives to teach people who truly care about learning and not stealing, through The School for Poetic Computation, openFrameworks, etc. It is difficult to share ideas without opening yourself up to be defrauded, but I am not sure where I stand on this yet.

In short, I'm not quite sure where I stand at any of it but that open source projects do foster incredible communities and that businesses should shift to truly support them.
